tools/roslaunch/src/roslaunch/xmlloader.py

```diff
@@ -223,6 +227,27 @@ def _check_attrs(self, tag, context, ros_config, attrs):
             if not t_a in attrs and not t_a in ['if', 'unless']:
                 ros_config.add_config_error("[%s] unknown <%s> attribute '%s'"%(context.filename, tag.tagName, t_a))
 
+    def _exec_fn_with_tmp_ctx(self, context, ros_config, fn):
+        """
+        Helper routine for executing functions in a context overwritten with ros config values.
+        @param context: LoaderContext
+        @param ros_config: launch configuration to load XML file into
+        @type  ros_config: L{ROSLaunchConfig}
+        @param fn: Generic function which takes context as its argument
+        @return: output of fn.
+        """
+        params_restore_point = copy.deepcopy(context.params)
+        # Use context to emulate get_param functionality in substitution_args
+        if ros_config.params:
+            if type(ros_config.params) is list:
+                for p in ros_config.params:
+                    context.add_param(p)
+
+        retval = fn(context)
+
+        context.params = copy.deepcopy(params_restore_point)
```

Comment:
Why does this need to b a `deepcopy` instead of a plain assignment?

Reply:
@tahsinkose Friendly ping.

Reply:
You are right. One `deepcopy` should suffice :+1: